src/Workspaces/Core/Portable/NamingStyles/NamingStyleRules.cs

```diff
@@ -1,13 +1,21 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
 {
     internal class NamingStyleRules
     {
         public ImmutableArray<NamingRule> NamingRules { get; }
 
+        private readonly ImmutableArray<SymbolKind> _symbolKindsThatCanBeOverridden =
```

Comment:
TBH, i don't see this array having any purpose.  YOu can just check the kind of the symbol in the method below.